vtex/faststore
# chore(file-system-api): drop search view

# What's the purpose of this PR?
This PR removes the search template from `gatsby-theme-store` and uses the Gatsby's File Sytem Route API for creating collection pages on the stores. Removing the search temaplate from `gatsby-theme-store` allowed us to do following breaking changes
1. (gatsby-source-vtex): Remove the staticPaths API
2. (gatsby-source-vtex); Remove Department sourcing
3. (gatsby-theme-store): Remove search template
4. (gatsby-theme-store): Drop createPage API all along

# Upgrading my store
To upgrade your store, you can use the changes made in our starter available at: https://github.com/vtex-sites/storecomponents.store/pull/1081.

TLDR: To have brand/cateogory pages, you need to create 3 pages with the following layout:
```
   | pages 
   -- | {StoreCollection.slug}
       -- | [...].tsx
       -- | index.tsx
   -- | s
       -- | [...].tsx
 ```
 
Now, a page will be created for each category/brand in your store. Also a `/s/:term` page will be created for full text searches.
You can use these newly created pages to implement the layout you desire

# How to test it?
This is just a code refactor and no visible feature should be changed. Make sure there are no regressions at:

https://github.com/vtex-sites/btglobal.store/pull/772
https://github.com/vtex-sites/marinbrasil.store/pull/585
